cset-workflow/app/fetch_fcst/bin/fetch-data-http.py

```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env python3
+
+"""Retrieve files via HTTP."""
+
+import ssl
+import urllib.parse
+import urllib.request
+
+from CSET._workflow_utils.fetch_data import FileRetriever, fetch_data
+
+
+class HTTPFileRetriever(FileRetriever):
+    """Retrieve files via HTTP."""
+
+    def get_file(self, file_path: str, output_dir: str) -> None:
+        """Save a file from a HTTP address to the output directory.
+
+        Parameters
+        ----------
+        file_path: str
+            Path of the file to copy on MASS. It may contain patterns
+            like globs, which will be expanded in a system specific manner.
+        output_dir: str
+            Path to filesystem directory into which the file should be copied.
+        """
+        ctx = ssl.create_default_context()
+        save_path = urllib.parse.urlparse(file_path).path.split("/")[-1]
+        with urllib.request.urlopen(file_path, output_dir, context=ctx) as response:
```

Comment:
This was more of a demonstration that implementing a very different data source was doable. I've done it this way to just stick to using the standard library, but we might say that requests is a common enough dependency that it is probably fine.

TBH, we will probably want it anyway if we add support for getting obs from APIs, which I know has been discussed before.

I think you are right here that I'm reading the entire response into memory before saving it, and would need to do chunked reading if I keep using this method.

Reply:
I've moved it into the fetch_data module, and have amended it to stream the data in 1 MiB chunks. I'm still using the stdlib `urllib.request` to avoid adding requests as a dependency, as nothing else needs it yet.